calyx/src/analysis/node_analysis.rs

```diff
@@ -0,0 +1,221 @@
+use crate::analysis::{DominatorMap, ReadWriteSet, ShareSet};
```

Comment:
Isn't this concept of `NODES` fundamentally tied to how dominator map works? If so, we should not expose this as a separate module and just provide the struct from the dominator map stuff. If you want different files for organizing purposes, create a new folder insider `analysis/` and expose all of the stuff from `analysis/dominator_map/mod.rs` in the way `analysis/mod.rs` does.